src/structures/static_stack.rs

```diff
@@ -0,0 +1,30 @@
+pub struct StaticStack<T: Sized + Default + Clone + Copy, const C: usize> {
+    buffer: [T; C],
+    top: usize,
+}
+
+impl<T: Sized + Default + Clone + Copy, const C: usize> StaticStack<T, C> {
+    pub fn new() -> Self {
+        Self {
+            buffer: [T::default(); C],
+            top: 0,
+        }
+    }
+
+    pub fn push(&mut self, item: &T) {
```

Comment:
should we do something about this panicking once it reaches the end of the stack or just let it panic?

Reply:
The purpose of the static stack is that you have to tell it how many slots it will have max - if it has too low I would say just let it panick and we will have to increase the size when it does